test/core/rules/rule_test_cases_test.py

```diff
@@ -20,10 +20,9 @@
 
     y = yaml.safe_load(raw)
 
-    rule = y["rule"]
-    named_tests = {k: v for k, v in y.items() if k != "rule"}
-    ids.extend([rule + "_" + t for t in named_tests])
-    test_cases.extend([RuleTestCase(rule=rule, **v) for k, v in named_tests.items()])
+    rule = y.pop("rule")
+    ids.extend([rule + "_" + t for t in y])
+    test_cases.extend([RuleTestCase(rule=rule, **v) for k, v in y.items()])
```

Comment:
@pwildenhain hope you don't mind me jumping on your PR but wanted to tidy this up before I forgot!